pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -67,7 +69,19 @@ func (e ShortcutExpander) getAll() []schema.GroupResource {
 		}
 	}
 
-	return availableAll
+	// populate all "discovered" resources from the server
+	for availableResource := range availableResources {
+		r := availableResource.GroupResource()
```

Comment:
@k82cn Thanks for the feedback, although I am using the index, in this case the map `availableResources` contains the actual groupVersionResource as the key, and an empty struct as the value